crates/db_views_moderator/src/modlog_combined_view.rs

```diff
@@ -0,0 +1,1405 @@
+      .into_boxed();
+
+    if let Some(mod_person_id) = self.mod_person_id {
+      query = query.filter(person::id.eq(mod_person_id));
+    };
+
+    if let Some(modded_person_id) = self.modded_person_id {
+      query = query.filter(modded_person.eq(modded_person_id));
+    };
+
+    if let Some(community_id) = self.community_id {
+      query = query.filter(community::id.eq(community_id))
+    }
+
+    if let Some(post_id) = self.post_id {
+      query = query.filter(post::id.eq(post_id))
+    }
+
+    if let Some(comment_id) = self.comment_id {
+      query = query.filter(comment::id.eq(comment_id))
+    }
+
+    if let Some(type_) = self.type_ {
+      query = match type_ {
+        ModlogActionType::All => query,
+        ModlogActionType::ModRemovePost => {
+          query.filter(modlog_combined::mod_remove_post_id.is_not_null())
+        }
+        ModlogActionType::ModLockPost => {
+          query.filter(modlog_combined::mod_lock_post_id.is_not_null())
+        }
+        ModlogActionType::ModFeaturePost => {
+          query.filter(modlog_combined::mod_feature_post_id.is_not_null())
+        }
+        ModlogActionType::ModRemoveComment => {
+          query.filter(modlog_combined::mod_remove_comment_id.is_not_null())
+        }
+        ModlogActionType::ModRemoveCommunity => {
+          query.filter(modlog_combined::mod_remove_community_id.is_not_null())
+        }
+        ModlogActionType::ModBanFromCommunity => {
+          query.filter(modlog_combined::mod_ban_from_community_id.is_not_null())
+        }
+        ModlogActionType::ModAddCommunity => {
+          query.filter(modlog_combined::mod_add_community_id.is_not_null())
+        }
+        ModlogActionType::ModTransferCommunity => {
+          query.filter(modlog_combined::mod_transfer_community_id.is_not_null())
+        }
+        ModlogActionType::ModAdd => query.filter(modlog_combined::mod_add_id.is_not_null()),
+        ModlogActionType::ModBan => query.filter(modlog_combined::mod_ban_id.is_not_null()),
+        ModlogActionType::ModHideCommunity => {
+          query.filter(modlog_combined::mod_hide_community_id.is_not_null())
+        }
+        ModlogActionType::AdminPurgePerson => {
+          query.filter(modlog_combined::admin_purge_person_id.is_not_null())
+        }
+        ModlogActionType::AdminPurgeCommunity => {
+          query.filter(modlog_combined::admin_purge_community_id.is_not_null())
+        }
+        ModlogActionType::AdminPurgePost => {
+          query.filter(modlog_combined::admin_purge_post_id.is_not_null())
+        }
+        ModlogActionType::AdminPurgeComment => {
+          query.filter(modlog_combined::admin_purge_comment_id.is_not_null())
+        }
+        ModlogActionType::AdminBlockInstance => {
+          query.filter(modlog_combined::admin_block_instance_id.is_not_null())
+        }
+        ModlogActionType::AdminAllowInstance => {
+          query.filter(modlog_combined::admin_allow_instance_id.is_not_null())
+        }
```

Comment:
You could make this easier to read by putting `use modlog_combined::*` just before the match. Maybe the compiler will give a warning then youd have to move the match into a separate function.